bliss/surveys/des.py

```diff
@@ -0,0 +1,465 @@
+                    image_basename = DESDownloader.image_basename_from_filename(image_id[bl], bl)
+                    self.download_image(brickname, image_id["sky_coord"], image_basename)
+
+    def download_image(self, brickname, sky_coord, image_basename):
+        """Download image for specified band, for this brick/ccd."""
+        image_table = self.svc.search((sky_coord["ra"], sky_coord["dec"])).to_table()
+        sel = defchararray.find(image_table["access_url"].astype(str), image_basename) != -1
+        b_access_url = image_table[sel][0]["access_url"]
+        image_dst_filename = self.download_dir / brickname[:3] / brickname / image_basename
+        try:
+            download_file_to_dst(b_access_url, image_dst_filename)
+        except HTTPError as e:
+            warnings.warn(
+                f"No {image_dst_filename} image for brick {brickname} at sky position "
+                f"({sky_coord['ra']}, {sky_coord['dec']}). Check cfg.datasets.des.bands.",
+                stacklevel=2,
+            )
+            raise e
+
+    def download_psfexs(self, bands: List[int]):
+        """Download psfex files for all image_ids."""
+        for image_id in self.band_image_filenames:
+            brickname = image_id["decals_brickname"]
+            for b, bl in enumerate(DES.BANDS):
+                if b in bands:
+                    self.download_psfex(brickname, image_id[bl])
+
+    def download_psfex(self, brickname, image_filename_no_ext):
+        """Download psfex file for specified band, for this brick/ccd."""
+        save_filename = self.save_filename_from_image_filename(image_filename_no_ext)
+        psfex_filename = (
+            self.download_dir / brickname[:3] / brickname / f"{save_filename}-psfex.fits"
+        )
+        download_file_to_dst(
+            f"{DESDownloader.URLBASE}/calib/psfex/{image_filename_no_ext}-psfex.fits",
+            psfex_filename,
+        )
+        return str(psfex_filename)
+
+    def download_backgrounds(self, bands: List[int]):
+        """Download sky params for all image_ids."""
+        for image_id in self.band_image_filenames:
+            brickname = image_id["decals_brickname"]
+            for b, bl in enumerate(DES.BANDS):
+                if b in bands:
+                    self.download_background(brickname, image_id[bl])
+
+    def download_background(self, brickname, image_filename_no_ext):
+        """Download sky params for specified band, for this brick/ccd."""
+        save_filename = self.save_filename_from_image_filename(image_filename_no_ext)
+        background_filename = (
+            self.download_dir / brickname[:3] / brickname / f"{save_filename}-splinesky.fits"
+        )
+        download_file_to_dst(
+            f"{DESDownloader.URLBASE}/calib/sky/{image_filename_no_ext}-splinesky.fits",
+            background_filename,
+        )
+        return str(background_filename)
+
+
+# NOTE: No DES catalog; re-use DecalsFullCatalog
+
+
+class DESPSF(ImagePSF):
```

Comment:
Maybe `DES_PSF` would be easier to parse? (And `SDSS_PSF`) too for that matter.

Reply:
Done.